### PR TITLE
INT-14843 - Fixed Students can't see "Awaiting Eula ?" to accept EULA after submission.

### PR DESCRIPTION
**Pull request by grijeshsaini**

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -214,15 +209,27 @@ public function get_links($linkarray) {
...
             if (!$moduleobject->show_other_posts_links($cm->course, $linkarray['userid'])) {
                 return $output;
             }
 
-            // Render the OR score or current submission status.
             if ($plagiarismfile) {
                 $submission = new plagiarism_turnitinsim_submission(new plagiarism_turnitinsim_request(), $plagiarismfile->id);
+            }
+
+            // If the user is a student and they are not allowed to view reports then return empty output.
```
Comment (jmcgettrick) [inline lib.php:224]: If I was being a pedant I would say expand on that comment but it's a pretty readable if statement.

File: lib.php [hunk 2]
```diff
@@ -214,15 +209,27 @@ public function get_links($linkarray) {
...
+            $plagiarismsettings = $this->get_settings($cm->id);
+            if (!$instructor && empty($plagiarismsettings->accessstudents) && $submission->getstatus() !== TURNITINSIM_SUBMISSION_STATUS_EULA_NOT_ACCEPTED) {
+                return $output;
+            }
+
+            // Render the OR score or current submission status.
+            if ($submission) {
+                //$submission = new plagiarism_turnitinsim_submission(new plagiarism_turnitinsim_request(), $plagiarismfile->id);
```
Comment (jmcgettrick) [inline lib.php:232]: Does that need to be removed?

File: lib.php [hunk 3]
```diff
@@ -214,15 +209,27 @@ public function get_links($linkarray) {
             // Get turnitin submission details.
             $plagiarismfile = plagiarism_turnitinsim_submission::get_submission_details($linkarray);
 
+            $submission = null;
```
Comment (jmcgettrick) [inline lib.php:212]: Can you move that up 4 lines please? Just to be next to all the other initialisations! 

Sorry for being pedantic! 😄

**Review by jmcgettrick — APPROVED** [review]

**Review by matdelong — APPROVED** [review]
LGTM

**Review by dwinn — APPROVED** [review]
Looks good to me.

**Pull request merged**